src/visibility.cpp

```diff
@@ -367,9 +367,17 @@ static bool rayLOSCallback(Vector2i pos, int32_t dist, void *data)
 			MAPTILE *psTile = mapTile(tile);
 			if (TileHasWall(psTile) && !TileHasSmallStructure(psTile))
 			{
-				help->lastHeight = 2 * UBYTE_MAX * ELEVATION_SCALE;
-				help->wall = pos.xy();
-				help->numWalls++;
+				STRUCTURE *psStruct = (STRUCTURE *)psTile->psObject;
+				if (psStruct->pStructureType->type == REF_GATE && psStruct->state == SAS_OPEN)
+				{
+					//Do not count open gates as blocking walls
+				}
+				else
+				{
```

Comment:
The condition could be changed to `psStruct->pStructureType->type != REF_GATE || psStruct->state != SAS_OPEN` , then we don't need the `else`.